File: Leetcode/1394 - Find Lucky Integer in an Array/solution.ts
function findLucky(arr: number[]): number {
    const numsCount: Record<number, number> = {};

    for (const num of arr) {
        numsCount[num] = (numsCount[num] ?? 0) + 1;
    }

    let res: number = -1;

    for (const key in numsCount) {
        const num: number = Number(key);
        if (numsCount[num] === num && num > res) {
            res = num;
        }
    }

    return res;
};

File: Leetcode/1394 - Find Lucky Integer in an Array/solution.py
def findLucky(self, arr: List[int]) -> int:
    numsCount: dict[int, int] = {}

    for num in arr:
        numsCount[num] = numsCount.get(num, 0) + 1

    res: int = -1

    for num in numsCount:
        if num == numsCount[num] and num > res:
            res = num

    return res